ang/civicase/DashboardTab.js

```diff
@@ -285,5 +289,59 @@
         return formattedResults;
       }
     }
+
+    /**
+     * Returns an object representing the "view cases" link
+     *
+     * Depending on the value of the relationship type filter, both the label
+     * and the url of the link might change
+     *
+     * This function is being called directly on the view so that the object
+     * is updated automatically whenever the relationship type filter value changes
+     *
+     * @return {Object}
+     */
+    function viewCasesLink () {
+      var queryParams = viewCasesQueryParams();
+
+      return {
+        url: $sce.trustAsResourceUrl('#/case/list?' + $.param(queryParams)),
+        label: $scope.filters.caseRelationshipType === 'all'
+          ? 'View all cases'
+          : 'View all my cases'
+      };
+    }
```

Comment:
I think we are not optimizing this call properly. Right now we have this:

```html
<a ng-href="{{customData.viewCasesLink().url}}">
  {{customData.viewCasesLink().label}}
</a>
```

But that means that each time we want to access a properly of the case link we generate it from scratch, calling `viewCasesQueryParams`, `$sce.trustAsResourceUrl`, etc.

It would be better if we watch for `$scope.filters.caseRelationshipType` changes and update a predefined object:

```js
$scope.viewCasesLink = {};

$scope.$watch('filters.caseRelationshipType', function () {
  $scope.viewCasesLink = getViewCasesLink();
});
```

A bit more complicated, but more optimized.